cdk/backend/function/createUserformBatch/index.js

```diff
@@ -429,24 +415,53 @@ async function createUserform(formDefinitionId) {
 }
 
 //Returns the X newest userforms by ownerId
-async function getUserformByOwnerWithLimit(limit = 1) {
+async function getUserformByOwnerWithLimit(formDefID = undefined, limit = 1) {
     try {
         let result = await docClient
             .query({
                 TableName: USERFORMTABLE_NAME,
                 IndexName: "byCreatedAt",
                 KeyConditionExpression: "#o = :ownerValue",
+                FilterExpression: "#formdefid = :formdefValue",
                 ExpressionAttributeNames: {
                     "#o": "owner",
+                    "#formdefid": "formDefinitionID"
                 },
                 ExpressionAttributeValues: {
                     ":ownerValue": ownerId,
+                    ":formdefValue": formDefID
                 },
                 ScanIndexForward: false, //Sort direction: true (default) = ASC, false = DESC
                 ConsistentRead: false,
                 Limit: limit,
             })
             .promise();
+        let lastEvaluatedKey = result.LastEvaluatedKey;
+        while (lastEvaluatedKey) {
```

Comment:
Her kunne vi vel kanskje gjenbrukt tankegangen med en funksjon som abstraherer vekk den while-loopen for å hente et fullt resultatsett, dersom dette er i bruk flere steder that is...hvis det bare er her er det ikke så farlig..